File: Dynamic Programming/sub set sum/python/subSetSum.py
def isSubsetSum(set,n, sum) : 
    
    # Base Cases 
    if (sum == 0) : 
        return True
    if (n == 0 and sum != 0) : 
        return False
   
    # If last element is greater than 
    # sum, then ignore it 
    if (set[n - 1] > sum) : 
        return isSubsetSum(set, n - 1, sum)
   
    # else, check if sum can be obtained 
    # by any of the following 
    # (a) including the last element 
    # (b) excluding the last element    
    return isSubsetSum(set, n-1, sum) or isSubsetSum(set, n-1, sum-set[n-1]) 
      
      
# Test case
set = [3, 34, 4, 12, 5, 2] 
sum = 100
n = len(set) 
if (isSubsetSum(set, n, sum) == True) : 
    print("Found a subset with given sum") 
else : 
    print("No subset with given sum") 

File: Dynamic Programming/sub set sum/Java/SubSetSum.java
class SubSetSum {  
    static boolean isSubsetSum(int set[], 
                               int n, int sum) 
    { 
        // Base Cases 
        if (sum == 0) 
            return true; 
        if (n == 0) 
            return false;  
        if (set[n - 1] > sum) 
            return isSubsetSum(set, n - 1, sum); 
        return isSubsetSum(set, n - 1, sum) 
            || isSubsetSum(set, n - 1, sum - set[n - 1]); 
    } 
  
    /* Driver program to test above function */
    public static void main(String args[]) 
    { 
        int set[] = { 3, 34, 4, 12, 5, 2 }; 
        int sum = 9; 
        int n = set.length; 
        if (isSubsetSum(set, n, sum) == true) 
            System.out.println("Subset found"
                               + " with given sum"); 
        else
            System.out.println("No subset with"
                               + " given sum"); 
    } 
}
